java/vector/src/main/codegen/templates/MapWriters.java

```diff
@@ -102,7 +116,7 @@ public Field getField() {
 
   @Override
   public MapWriter map(String name) {
-      FieldWriter writer = fields.get(name.toLowerCase());
+      FieldWriter writer = fields.get(handleCase(name));
```

Comment:
@wesm @jacques-n I am a bit confused here as i don't have the full picture of how these writer classes are used. What is the benefit of subclassing into CaseSensitiveMapWriter?